pkg/reconciler/v1alpha1/trigger/resources/subscription.go

```diff
@@ -48,11 +50,7 @@ func NewSubscription(t *eventingv1alpha1.Trigger, brokerTrigger, brokerIngress *
 				Name:       brokerTrigger.Name,
 			},
 			Subscriber: &eventingv1alpha1.SubscriberSpec{
-				Ref: &corev1.ObjectReference{
-					APIVersion: "v1",
-					Kind:       "Service",
-					Name:       svc.Name,
-				},
+				URI: &uriString,
```

Comment:
Foodforthought: Based on the discussions on Addressabel, it seems like converting this to a url.URL will make it extensible in future. See the suggestion that [Scott gave](https://github.com/knative/pkg/pull/362) 
Could you check if it will be a breaking change or is it something we could do as part of this change?

Reply:
I don't see Scott's comment.

My memory is that all the recommendations have still serialized the URL as a string, which this does. If we change the type in Go code, as long as it still serializes to a string, I think we can make the change in the future. Serving seems to just use strings, see https://github.com/knative/serving/blob/5be3e3e80062c3d43751aa33317206ca45ce6179/pkg/apis/serving/v1alpha1/route_types.go#L89.